pkg/reconcile/pipeline/infinispan/handler/provision/gossiprouter.go

```diff
@@ -129,33 +129,43 @@ func GossipRouter(i *ispnv1.Infinispan, ctx pipeline.Context) {
 		}
 
 		router.Labels = routerLabels
+
+		container := &corev1.Container{
+			Name:           GossipRouterContainer,
+			Image:          i.ImageName(),
+			Command:        []string{"/opt/gossiprouter/bin/launch.sh"},
+			Args:           args,
+			Env:            []corev1.EnvVar{{Name: "ROUTER_JAVA_OPTIONS", Value: i.Spec.Container.RouterExtraJvmOpts}},
+			Ports:          containerPorts,
+			LivenessProbe:  TcpProbe(probePort, 10, 5, 0, 1, 60),
+			ReadinessProbe: TcpProbe(probePort, 10, 5, 0, 1, 60),
+			StartupProbe:   TcpProbe(probePort, 15, 1, 1, 1, 60),
+		}
+
+		if podResources, err := gossipRouterPodResources(i.Spec.Service.Sites.Local.Discovery); err != nil {
+			ctx.Requeue(fmt.Errorf("unable to calculate Gossip Router pod resources: %w", err))
```

Comment:
As this is in the mutateFn, should we just be returning the error here so that the `CreateOrUpdate` call fails with "Failed to configure Cross-Site Deployment..."